### ISSUE TITLE
SSR: "useLayoutEffect does nothing on the server"

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/atomic-layout#293]
## What:

During the SSR there is the following exception being thrown:

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://fb.me/react-uselayouteffect-ssr for common fixes.
    in c
    in Fragment
    in Placeholder(styled.div)
    in Unknown (at Header.tsx:51)
    in header
```

## Why:

It's related to the usage of `useLayoutEffect` by one of the library's hooks. The adoption of that hooks was required to fix the "jumping" layout issue (#234).

## Environment:

- node vesrion: `v10.16.3`
- npm version: `6.9.0`
- atomic-layout version: `0.13.0`

## Steps to reproduce:

1. Create a `Composition` component with arbitrary areas.
1. Render it on a server.

## Expected behavior:

There is no exception being thrown during server-side rendering.

**Comment by kettanaito** [comment]
I will try to tackle this by conditionally deciding which effect hook to use:

- `useEffect` for the server
- `useLayoutEffect` for the client (see #254)

**Comment by kettanaito** [comment]
Published in `0.12.6`.
